### PR TITLE
WIP. [NTOS:KDBG] KdbpInNmiOrDoubleFaultHandler(): Fix checks. CORE-16448

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Addendum to r54503.
Addendum to 3c8f19eb2153034b9b47fc91284b54b1200f21d8.

JIRA issue: [CORE-16448](https://jira.reactos.org/browse/CORE-16448)

### CODE REVIEW

**Comment by SergeGautherie** [comment]
Cc @HBelusca, for your regression part only.

**Comment by HBelusca** [comment]
That's a good question, indeed. Cc @hpoussin too. My question here is why is it better to do these comparisons with respect to the C handlers entry points instead of the corresponding ASM ones?

**Comment by SergeGautherie** [comment]
> My question

+1.
I assumed the 1 case in #1996 and the 1+1 here should all work similarly.
If that is not the case, tell me which case should check what (ASM, C, both). Thanks.

File: ntoskrnl/kdbg/kdb_cli.c [hunk 1]
```diff
@@ -1051,17 +1051,17 @@ KdbpTrapFrameFromPrevTss(
...
 VOID FASTCALL KiTrap03Handler(IN PKTRAP_FRAME);
-VOID __cdecl KiTrap08(VOID);
-VOID __cdecl KiTrap09(VOID);
+VOID __cdecl  KiTrap08Handler(VOID);
+VOID FASTCALL KiTrap09Handler(IN PKTRAP_FRAME);
 
 static BOOLEAN
 KdbpInNmiOrDoubleFaultHandler(
-    ULONG_PTR Address)
+    PVOID Address)
```
Comment (learn-more) [inline ntoskrnl/kdbg/kdb_cli.c]: Why?

File: ntoskrnl/kdbg/kdb_cli.c [hunk 2]
```diff
@@ -1171,7 +1171,7 @@ KdbpCmdBackTrace(
         if (Address == 0)
             break;
 
-        if (KdbpInNmiOrDoubleFaultHandler(Address))
+        if (KdbpInNmiOrDoubleFaultHandler((PVOID)Address))
```
Comment (learn-more) [inline ntoskrnl/kdbg/kdb_cli.c]: Why?

File: ntoskrnl/kdbg/kdb_cli.c (hunk 1 above)
Comment (SergeGautherie) [reply ntoskrnl/kdbg/kdb_cli.c]: Same pattern as #1996, easier to read, ...

File: ntoskrnl/kdbg/kdb_cli.c (hunk 2 above)
Comment (SergeGautherie) [reply ntoskrnl/kdbg/kdb_cli.c]: Cast is required here now!

File: ntoskrnl/kdbg/kdb_cli.c (hunk 1 above)
Comment (binarymaster) [reply ntoskrnl/kdbg/kdb_cli.c]: > easier to read

Totally not.

**Comment by SergeGautherie** [comment]
> There is no guarantee

I concur. Yet, my initial goal is to fix the regression (and improve a bit).
If a better fix/rewrite is not trivial/easy, then that should be a follow-up issue/PR.

> Only the ASM handler is special.

I believe I understand/agree with that.
Is the change you are asking for to check ASM function addresses only, instead?
If not, then could you clarify what you expect? Thanks.

**Comment by HBelusca** [comment]
Superseded by commit d21ff0ed137d61b7881af96dd1d6e14387ccb89b .

**Pull request closed without merging**